src/lib/features/playground/playground-service.ts

```diff
@@ -29,23 +38,38 @@ export class PlaygroundService {
         this.segmentService = segmentService;
     }
 
-    async evaluateQuery(
+    async evaluateAdvancedQuery(
         projects: typeof ALL | string[],
-        environment: string,
+        environments: string[],
         context: SdkContextSchema,
     ): Promise<PlaygroundFeatureSchema[]> {
-        const [features, segments] = await Promise.all([
-            this.featureToggleService.getClientFeatures(
-                {
-                    project: projects === ALL ? undefined : projects,
-                    environment,
-                },
-                true,
-                false,
+        const segments = await this.segmentService.getActive();
+        const environmentFeatures = await Promise.all(
+            environments.map((env) => this.resolveFeatures(projects, env)),
+        );
+        const contexts = generateObjectCombinations(context);
+
+        const results = await Promise.all(
```

Comment:
this will be thoroughly tested in subsequent PRs. Here I wrote this to manually test the combinatorial generation and drive the extract method refactoring